examples/osm-vector-tiles.js

```diff
@@ -3,58 +3,60 @@ import TopoJSON from '../src/ol/format/TopoJSON.js';
 import VectorTileLayer from '../src/ol/layer/VectorTile.js';
 import VectorTileSource from '../src/ol/source/VectorTile.js';
 import View from '../src/ol/View.js';
-import {Fill, Stroke, Style} from '../src/ol/style.js';
 import {fromLonLat} from '../src/ol/proj.js';
 
 const key = 'uZNs91nMR-muUTP99MyBSg';
 
-const roadStyleCache = {};
-const roadColor = {
-  'major_road': '#776',
-  'minor_road': '#ccb',
-  'highway': '#f39',
-};
-const buildingStyle = new Style({
-  fill: new Fill({
-    color: '#666',
-    opacity: 0.4,
-  }),
-  stroke: new Stroke({
-    color: '#444',
-    width: 1,
-  }),
-});
-const waterStyle = new Style({
-  fill: new Fill({
-    color: '#9db9e8',
-  }),
-});
-const roadStyle = function (feature) {
-  const kind = feature.get('kind');
-  const railway = feature.get('railway');
-  const sort_key = feature.get('sort_key');
-  const styleKey = kind + '/' + railway + '/' + sort_key;
-  let style = roadStyleCache[styleKey];
-  if (!style) {
-    let color, width;
-    if (railway) {
-      color = '#7de';
-      width = 1;
-    } else {
-      color = roadColor[kind];
-      width = kind == 'highway' ? 1.5 : 1;
-    }
-    style = new Style({
-      stroke: new Stroke({
-        color: color,
-        width: width,
-      }),
-      zIndex: sort_key,
-    });
-    roadStyleCache[styleKey] = style;
-  }
-  return style;
-};
+const rules = [
+  {
+    filter: ['==', ['get', 'layer'], 'water'],
+    style: {
+      'fill-color': '#9db9e8',
+    },
+  },
+  {
+    else: true,
+    filter: ['all', ['==', ['get', 'layer'], 'roads'], ['get', 'railway']],
+    style: {
+      'stroke-color': '#7de',
+      'stroke-width': 1,
+      'z-index': ['number', ['get', 'sort_key'], 0],
+    },
+  },
+  {
+    else: true,
+    filter: ['==', ['get', 'layer'], 'roads'],
+    style: {
+      'stroke-color': [
+        'match',
+        ['get', 'kind'],
+        'major_road',
+        '#776',
+        'minor_road',
+        '#ccb',
+        'highway',
+        '#f39',
+        'none',
```

Comment:
I decided to build on the logic for handling `none` as a fill color.  The canvas style builder will now return `null` for a stroke if the color evaluates to `none`.